src/Image/ResizeConfiguration.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Image;
+
+/**
+ * Resize configuration used by the ResizeCalculator
+ *
+ * @author Martin Auswöger <martin@auswoeger.com>
+ */
+class ResizeConfiguration
+{
+    /**
+     * @var int
+     */
+    private $width = 0;
+
+    /**
+     * @var int
+     */
+    private $height = 0;
+
+    /**
+     * @var string
+     */
+    private $mode = self::MODE_CROP;
+
+    const MODE_CROP = 'crop';
+    const MODE_BOX = 'box';
+    const MODE_PROPORTIONAL = 'proportional';
+
+    /**
+     * @var int
+     */
+    private $zoomLevel = 0;
+
+    /**
+     * Gets the width.
+     *
+     * @return int
+     */
+    public function getWidth()
+    {
+        return $this->width;
+    }
+
+    /**
+     * Sets the width.
+     *
+     * @param int $width the width
+     *
+     * @return self
+     */
+    public function setWidth($width)
+    {
+        $width = (int) $width;
+
+        if ($width < 0) {
+            throw new \InvalidArgumentException('Width must not be negative');
+        }
+
+        $this->width = $width;
+
+        return $this;
+    }
+
+    /**
+     * Gets the height.
+     *
+     * @return int
+     */
+    public function getHeight()
+    {
+        return $this->height;
+    }
+
+    /**
+     * Sets the height.
+     *
+     * @param int $height the height
+     *
+     * @return self
+     */
+    public function setHeight($height)
+    {
+        $height = (int) $height;
```

Comment:
I don’t like that, because e.g. `-0.1` would than throw an error but `(int)-0.1` wouldn’t. In more complex methods, casting after the validation could be used to bypass the validation.